### PR TITLE
Just Troubleshoot

### PR DESCRIPTION
**Pull request by andyneff**
Added new just troubleshoot feature. In your main project, you can:

1. Create a `.troubleshoot` file that points to other troubleshoot tests
2. Create a `.troubleshoot.d` directory containing troubleshoot tests

A troubleshoot test file contains pairs of functions:

- `troubleshoot_{test_name}` - Test to run (under `set -eu`), returns true when everything is ok. False when an issue is detected
- `error_{test_name}` - Prints _helpful_ error message when troubleshoot test fails

### CODE REVIEW

File: tests/troubleshoot/nvidia_docker [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+  docker run -it --rm --gpus=all "${TERRA_WRIVA_BASE_IMAGE}" nvidia-smi
+}
+
+function error_nvidia_docker()
+{
+  echo "${RED}NVIDIA docker does not appear to be working. If you have NVIDIA"
+  echo "GPUs that you expect to be working, you may have 'no-cgroups=true' in"
+  echo "/etc/nvidia-container-runtime/config.toml. Either have an admin revert"
+  echo "this to false, or set ${YELLOW}TERRA_WRIVA_PRIVILEGED_MODE=true"
+  echo "${RED}in your ${YELLOW}local.env ${RED}file.${NC}"
```
Comment (scott-vsi) [inline tests/troubleshoot/nvidia_docker:14]:
```suggestion
  echo "${RED}in your ${YELLOW}local.env ${RED}file.${NC}", which will run the
  container in privileged mode.
```

File: linux/troubleshoot [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+      end_test
+      if [ "${test_status}" != "0" ]; then
+        error_name=error_${test_name#troubleshoot_}
+        "${error_name}"
+      fi
+    done
+    TESTLIB_TEST_FILENAME=$(basename "${1}") atexit
+  ) || :
+  else
+    echo "Skipping ${1}, cannot read"
```
Comment (scott-vsi) [inline linux/troubleshoot:37]: Is there any more information to give here? like cannot fine troubleshoot file or something

Comment (andyneff) [reply linux/troubleshoot:37]: There are multiple reasons why it cannot read. It could not exist, be a bad symlink, exit but not have read permissions. Etc. This is why I specifically test `-r` cause this tests all these corner cases.

**Review by scott-vsi — APPROVED** [review]
